File: Data Structures & Algorithms/minimum-stack/submission-2.java
class MinStack {
    //minheap
    PriorityQueue<Integer> minHeap;
    LinkedList<Integer> numbers;
    public MinStack() {
        numbers=new LinkedList<>();
        minHeap=new PriorityQueue<>();
    }
    
    public void push(int val) {
        numbers.push(val);
        if(minHeap.isEmpty()||val<=minHeap.peek())
        minHeap.offer(val);
    }
    
    public void pop() {
        if(numbers.isEmpty()) return;
        int n=numbers.pop();
        if(minHeap.peek()==n){
            minHeap.poll();
        }
    }
    
    public int top() {
        return numbers.peek();
    }
    
    public int getMin() {
        return minHeap.peek();
    }
}

File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack=[]
        for c in s:
            if c=="{" or c=="(" or c=="[":
                stack.append(c)
            else:
                if not stack:
                    return False 
                peek=stack.pop()
                if (peek=="[" and c!="]") or peek=="{" and c!="}" or peek=="(" and c!=")":
                    return False
        return True if not stack else False